azurelinuxagent/common/protocol/wire.py

```diff
@@ -623,103 +596,141 @@ def call_storage_service(http_req, *args, **kwargs):
 
-        response = self._fetch_response(uri, headers, use_proxy,  max_retry=max_retry)
-        if response is not None and not restutil.request_failed(response):
-            chunk_size = 1024 * 1024  # 1MB buffer
-            try:
+            response = self._fetch_response(uri, headers, use_proxy)
+            if response is not None and not restutil.request_failed(response):
+                chunk_size = 1024 * 1024  # 1MB buffer
                 with open(destination, 'wb', chunk_size) as destination_fh:
                     complete = False
                     while not complete:
                         chunk = response.read(chunk_size)
                         destination_fh.write(chunk)
                         complete = len(chunk) < chunk_size
-                success = True
-            except Exception as error:
-                logger.error('Error streaming {0} to {1}: {2}'.format(uri, destination, ustr(error)))
-
-        return success
+            return ""
+        except:
+            if os.path.exists(destination):  # delete the destination file, in case we did a partial download
+                try:
+                    os.remove(destination)
+                except Exception as exception:
+                    logger.warn("Can't delete {0}: {1}", destination, ustr(exception))
+            raise
 
-    def fetch(self, uri, headers=None, use_proxy=None, decode=True, max_retry=None, retry_codes=None, ok_codes=None):
+    def fetch(self, uri, headers=None, use_proxy=None, decode=True, retry_codes=None, ok_codes=None):
         """
-        max_retry indicates the maximum number of retries for the HTTP request; None indicates that the default value should be used
-
         Returns a tuple with the content and headers of the response. The headers are a list of (name, value) tuples.
         """
         logger.verbose("Fetch [{0}] with headers [{1}]", uri, headers)
         content = None
         response_headers = None
-        response = self._fetch_response(uri, headers, use_proxy, max_retry=max_retry, retry_codes=retry_codes, ok_codes=ok_codes)
+        response = self._fetch_response(uri, headers, use_proxy, retry_codes=retry_codes, ok_codes=ok_codes)
         if response is not None and not restutil.request_failed(response, ok_codes=ok_codes):
             response_content = response.read()
             content = self.decode_config(response_content) if decode else response_content
             response_headers = response.getheaders()
         return content, response_headers
 
-    def _fetch_response(self, uri, headers=None, use_proxy=None, max_retry=None, retry_codes=None, ok_codes=None):
-        """
-        max_retry indicates the maximum number of retries for the HTTP request; None indicates that the default value should be used
-        """
+    def _fetch_response(self, uri, headers=None, use_proxy=None, retry_codes=None, ok_codes=None):
```

Comment:
removed the max_retry parameter, since it is no longer used. All downloads have the same strategy in terms of timeouts, retries and error reporting